### PR TITLE
Authentication for admin endpoints through kubernetes ingress

### PR DESCRIPTION
**Pull request by izzyblues**

### CODE REVIEW

File: helm/upp-aggregate-healthcheck/templates/admin-ingress.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{{ if eq .Values.ingress.enabled "true" }}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{.Values.service.name}}-admin-ingress
```
Comment (peteclark-ft) [inline helm/upp-aggregate-healthcheck/templates/admin-ingress.yaml]: The admin-ingress' are the `/__service-name` style endpoints, so I'd rename this

File: helm/upp-aggregate-healthcheck/templates/admin-ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    # name of the secret that contains the user/password definitions
+    ingress.kubernetes.io/auth-secret: basic-auth
+    # message to display with an appropiate context why the authentication is required
+    ingress.kubernetes.io/auth-realm: "Authentication Required"
+spec:
+  rules:
+    - host: "*.ft.com"
+      http:
+        paths:
+        - path: ~ {{ .Values.env.PATH_PREFIX }}/(?!__).*
```
Comment (peteclark-ft) [inline helm/upp-aggregate-healthcheck/templates/admin-ingress.yaml]: Hmm this isn't perfect as `/__health/` will now require auth, when it shouldn't... Can we do better?

Comment (peteclark-ft) [reply helm/upp-aggregate-healthcheck/templates/admin-ingress.yaml]: `/(?!__).+` maybe?

File: helm/upp-aggregate-healthcheck/templates/admin-ingress.yaml (hunk 1 above)
Comment (izzyblues) [reply helm/upp-aggregate-healthcheck/templates/admin-ingress.yaml]: Renamed as operation-ingrees

File: helm/upp-aggregate-healthcheck/templates/admin-ingress.yaml (hunk 2 above)
Comment (izzyblues) [reply helm/upp-aggregate-healthcheck/templates/admin-ingress.yaml]: Yes! Your regex fixes the problem

**Pull request merged**